### PR TITLE
Add mission raw plugin

### PR DESCRIPTION
**Pull request by julianoes**
This will be a plugin which enables direct/raw access to MAVLink MISSION_ITEM_INT messages.

See progress here @sanderux.

### CODE REVIEW

File: plugins/mission/mission.cpp [hunk 1]
```diff
@@ -65,18 +65,20 @@ const char *Mission::result_str(Result result)
...
             return "Success";
-        case Result::BUSY:
-            return "Busy";
         case Result::ERROR:
             return "Error";
         case Result::TOO_MANY_MISSION_ITEMS:
             return "Too many mission items";
-        case Result::INVALID_ARGUMENT:
-            return "Invalid argument";
+        case Result::BUSY:
```
Comment (JonasVautherin) [inline plugins/mission/mission.cpp]: What's the link between `mission.cpp` and the `mission_raw` plugin?

Comment (sanderux) [reply plugins/mission/mission.cpp]: @JonasVautherin the mission plugin only returns values known to the SDK. The raw plugin returns all of them.

Comment (JonasVautherin) [reply plugins/mission/mission.cpp]: Right, but I see a `Result` defined in `plugins/mission_raw`. This one is from `plugins/mission`, which is a different plugin. Do we need those changes in the `mission` plugin, too? And is that part of this PR?

Comment (julianoes) [reply plugins/mission/mission.cpp]: Oh right. I just found that one case was missing as part of copying from `mission`.

File: integration_tests/mission_raw_mission_changed.cpp [hunk 2]
```diff
@@ -2,11 +2,19 @@
...
 #include <future>
 
 using namespace dronecode_sdk;
-using namespace std::placeholders; // for `_1`
+
+static constexpr double SOME_LATITUDES[] = {47.398170, 47.398175};
+static constexpr double SOME_LONGITUDES[] = {8.545649, 8.545654};
+static constexpr float SOME_ALTITUDES[] = {5.0f, 7.5f};
+static constexpr float SOME_SPEEDS[] = {4.0f, 5.0f};
+static constexpr unsigned NUM_SOME_ITEMS = sizeof(SOME_LATITUDES) / sizeof(SOME_LATITUDES[0]);
```
Comment (JonasVautherin) [inline integration_tests/mission_raw_mission_changed.cpp]: Could we use a `vector` instead? Would be clearer to me :confused:

File: integration_tests/mission_raw_mission_changed.cpp [hunk 3]
```diff
@@ -34,30 +42,67 @@ TEST_F(SitlTest, MissionRawMissionChanged)
...
+
+    {
+        std::promise<void> prom{};
+        std::future<void> fut = prom.get_future();
+        LogInfo() << "Download raw mission items.";
+        mission_raw->download_mission_async(
+            [&prom](MissionRaw::Result result,
+                    std::vector<std::shared_ptr<MissionRaw::MavlinkMissionItemInt>> items) {
+                EXPECT_EQ(result, MissionRaw::Result::SUCCESS);
+                // TODO: validate items
```
Comment (JonasVautherin) [inline integration_tests/mission_raw_mission_changed.cpp]: That TODO is outdated, isn't it?

**Comment by julianoes** [comment]
I'm ignoring Jenkins because I don't know what he's trying to tell me.

**Comment by julianoes** [comment]
@JonasVautherin also I forgot to fix what you raised, I'll do that in #642.

File: plugins/mission_raw/include/plugins/mission_raw/mission_raw.h [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+     *
+     * @param result Enum for which string is required.
+     * @return Human readable string for the MissionRaw::Result.
+     */
+    static const char *result_str(Result result);
+
+    /**
+     * @brief Mission item identical to MAVLink MISSION_ITEM_INT.
+     */
+    struct MavlinkMissionItemInt {
```
Comment (hamishwillee) [inline plugins/mission_raw/include/plugins/mission_raw/mission_raw.h]: No support for MISSION_ITEM?

Comment (julianoes) [reply plugins/mission_raw/include/plugins/mission_raw/mission_raw.h]: Nope. The SDK only does the new shiny things.

**Pull request merged**